src/main/java/org/folio/orders/utils/PoLineCommonUtil.java

```diff
@@ -125,29 +119,41 @@ public static boolean isItemsUpdateRequired(CompositePoLine compPOL) {
   }
 
   public static boolean isItemsUpdateRequiredForEresource(CompositePoLine compPOL) {
-    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
-    if (!(format == ELECTRONIC_RESOURCE || format == P_E_MIX))
-      return false;
     if (compPOL.getCheckinItems() != null && compPOL.getCheckinItems()) {
       return false;
     }
-    return Optional.ofNullable(compPOL.getEresource())
+    return Optional.ofNullable(getEresource(compPOL))
       .map(eresource -> eresource.getCreateInventory() == Eresource.CreateInventory.INSTANCE_HOLDING_ITEM)
       .orElse(false);
   }
 
   public static boolean isItemsUpdateRequiredForPhysical(CompositePoLine compPOL) {
-    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
-    if (!(format == PHYSICAL_RESOURCE || format == OTHER || format == P_E_MIX))
-      return false;
     if (compPOL.getCheckinItems() != null && compPOL.getCheckinItems()) {
       return false;
     }
-    return Optional.ofNullable(compPOL.getPhysical())
+    return Optional.ofNullable(getPhysical(compPOL))
       .map(physical -> physical.getCreateInventory() == Physical.CreateInventory.INSTANCE_HOLDING_ITEM)
       .orElse(false);
   }
 
+  public static Physical getPhysical(PoLine poLine) {
+    return getPhysical(convertToCompositePoLine(poLine));
+  }
+
+  public static Eresource getEresource(PoLine poLine) {
+    return getEresource(convertToCompositePoLine(poLine));
+  }
+
+  public static Physical getPhysical(CompositePoLine compPOL) {
+    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
+    return format == PHYSICAL_RESOURCE || format == P_E_MIX || format == OTHER ? compPOL.getPhysical() : null;
+  }
+
+  public static Eresource getEresource(CompositePoLine compPOL) {
+    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
+    return format == ELECTRONIC_RESOURCE || format == P_E_MIX ? compPOL.getEresource() : null;
+  }
+
   public static boolean isOnlyInstanceUpdateRequired(CompositePoLine compPOL) {
```

Comment:
I still believe that it should be in this format because making method closer good strategy. Reminder it is suggestion, you can decide
```
  public static Eresource getEresource(PoLine poLine) {
    return getEresource(convertToCompositePoLine(poLine));
  }
  
  public static Eresource getEresource(CompositePoLine compPOL) {
    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
    return format == ELECTRONIC_RESOURCE || format == P_E_MIX ? compPOL.getEresource() : null;
  }
  
  public static Physical getPhysical(PoLine poLine) {
    return getPhysical(convertToCompositePoLine(poLine));
  }
  
  public static Physical getPhysical(CompositePoLine compPOL) {
    CompositePoLine.OrderFormat format = compPOL.getOrderFormat();
    return format == PHYSICAL_RESOURCE || format == P_E_MIX || format == OTHER ? compPOL.getPhysical() : null;
  }
```

Reply:
done